src/tasks/LargeFileUploadTask.ts

```diff
@@ -0,0 +1,344 @@
+    await this.updateSession();
+    return this.upload(progress);
+  }
+
+  /**
+   * Refreshes the current upload session status by making a GET request to the upload URL.
+   * Updates the session expiration date, next expected ranges, and remaining ranges based on the response.
+   *
+   * @returns {Promise<UploadSession | undefined>} - A promise that resolves to the updated upload session.
+   * @throws {Error} If the request fails.
+   */
+  public async updateSession(): Promise<UploadSession | undefined> {
+    const url = this.Session.uploadUrl;
+    if (!url) {
+      throw new Error("Upload url is invalid");
+    }
+    const requestInformation = new RequestInformation(HttpMethod.GET, url);
+    const response = await this.requestAdapter.send<UploadSession>(
+      requestInformation,
+      createUploadSessionFromDiscriminatorValue,
+      this.errorMappings,
+    );
+
+    if (response) {
+      this.Session.expirationDateTime = response.expirationDateTime;
+      this.Session.nextExpectedRanges = response.nextExpectedRanges;
+      if (response.uploadUrl) {
+        this.Session.uploadUrl = response.uploadUrl;
+      }
+      this.rangesRemaining = this.getRangesRemaining(this.Session);
+    }
+    return response;
+  }
+
+  /**
+   * Deletes the current upload session.
+   * Sends a PUT request to the upload URL to cancel the session.
+   *
+   * @returns {Promise<void>} A promise that resolves when the session is canceled.
+   */
+  public async deleteSession(): Promise<void> {
+    const url = this.Session.uploadUrl;
+    if (!url) {
+      throw new Error("Upload url is invalid");
+    }
+    const requestInformation = new RequestInformation(HttpMethod.DELETE, url);
+    await this.requestAdapter.sendNoResponseContent(requestInformation, this.errorMappings);
+  }
+
+  /**
+   * Extracts the upload session information from a parsable object.
+   *
+   * @param {Parsable} parsable - The parsable object containing the upload session information.
+   * @returns {UploadSession} - The extracted upload session information.
+   */
+  private extractSessionInfo(parsable: Parsable): UploadSession {
+    const { expirationDateTime, nextExpectedRanges, odataType, uploadUrl } = parsable as Partial<UploadSession>;
+    if (!nextExpectedRanges || !uploadUrl || nextExpectedRanges.length === 0) {
+      throw new Error("Upload session is invalid");
+    }
+    return {
+      expirationDateTime: expirationDateTime ?? null,
+      nextExpectedRanges: nextExpectedRanges ?? null,
+      odataType: odataType ?? null,
+      uploadUrl: uploadUrl ?? null,
+    };
+  }
+
+  /**
+   * Generates a list of upload slice requests based on the remaining ranges to be uploaded.
+   *
+   * @private
+   * @returns {UploadSlice<T>[]} An array of UploadSlice objects representing the upload requests.
+   */
+  private getUploadSliceRequests(): UploadSlice<T>[] {
+    const uploadSlices: UploadSlice<T>[] = [];
```

Comment:
here we're still getting all the remaining slices, instead of just the next one.
I think we need also update that to avoid memory pressure